### PR TITLE
Use crossbeam channel to replace std::channel

### PR DESCRIPTION
**Pull request by yangby-cryptape**

### CODE REVIEW

**Comment by kaikai1024** [comment]
> If we want to use crossbean channel, we must use pubsub, It's a little weird.

NO. if you want to use `start_pubseb`, you need to use `pubsub`, you can use crossbeam as you want.

File: cita-network/src/connection.rs [hunk 1]
```diff
@@ -23,14 +23,15 @@ use libproto::TryInto;
...
+use pubsub::channel::{self, Receiver, Sender};
 use std::collections::HashMap;
 use std::fs::File;
 use std::io;
 use std::io::{Read, Write};
 use std::net::{Shutdown, TcpStream};
 use std::net::{SocketAddr, ToSocketAddrs};
 use std::sync::atomic::{AtomicBool, AtomicUsize, Ordering};
-use std::sync::mpsc::{channel, Receiver, Sender};
+use std::sync::mpsc::Receiver as StdReceiver;
```
[the remark this answers is not in the document]
Comment (yangby-cryptape) [reply cita-network/src/connection.rs]: [A third-part crate](https://crates.io/crates/notify) need [a `std::sync::mpsc::Receiver` as input](https://docs.rs/notify/4.0.6/notify/trait.Watcher.html#tymethod.new).

https://github.com/cryptape/cita/blob/dd3e0cea53b9f96c3bcd9d387d5eead7042646d1/cita-network/Cargo.toml#L14
https://github.com/cryptape/cita/blob/dd3e0cea53b9f96c3bcd9d387d5eead7042646d1/Cargo.lock#L572
https://github.com/cryptape/cita/blob/dd3e0cea53b9f96c3bcd9d387d5eead7042646d1/cita-network/src/main.rs#L120
https://github.com/cryptape/cita/blob/dd3e0cea53b9f96c3bcd9d387d5eead7042646d1/cita-network/src/main.rs#L196-L198

[the remark this answers is not in the document]
Comment (u2) [reply cita-network/src/connection.rs]: I see.

**Comment by yangby-cryptape** [comment]
> If we want to use crossbean channel, we must use pubsub, It's a little weird.

I prefer to say that: if we want to use the same channel as `pubsub`, we can use `pubsub::channel`.

If we want use another `channel`, that is OK, for example:

https://github.com/cryptape/cita/blob/dd3e0cea53b9f96c3bcd9d387d5eead7042646d1/cita-executor/src/main.rs#L176-L181

https://github.com/cryptape/cita/blob/9e25a72c347f5e9c9b820af9689306e442018fba/cita-network/src/connection.rs#L406

**Comment by u2** [comment]
> > If we want to use crossbean channel, we must use pubsub, It's a little weird.
> 
> I prefer to say that: if we want to use the same channel as `pubsub`, we can use `pubsub::channel`.
> 
> If we want use another `channel`, that is OK, for example:
> 
> [cita/cita-executor/src/main.rs](https://github.com/cryptape/cita/blob/dd3e0cea53b9f96c3bcd9d387d5eead7042646d1/cita-executor/src/main.rs#L176-L181)
> 
> Lines 176 to 181 in [dd3e0ce](/cryptape/cita/commit/dd3e0cea53b9f96c3bcd9d387d5eead7042646d1)
> 
>  let (mq_req_sender, mq_req_receiver) = crossbeam_channel::unbounded(); 
>  let (mq_resp_sender, mq_resp_receiver) = crossbeam_channel::unbounded(); 
>  let (fsm_req_sender, fsm_req_receiver) = crossbeam_channel::unbounded(); 
>  let (fsm_resp_sender, fsm_resp_receiver) = crossbeam_channel::unbounded(); 
>  let (command_req_sender, command_req_receiver) = crossbeam_channel::bounded(0); 
>  let (command_resp_sender, command_resp_receiver) = crossbeam_channel::bounded(0); 
> [cita/cita-network/src/connection.rs](https://github.com/cryptape/cita/blob/9e25a72c347f5e9c9b820af9689306e442018fba/cita-network/src/connection.rs#L406)
> 
> Line 406 in [9e25a72](/cryptape/cita/commit/9e25a72c347f5e9c9b820af9689306e442018fba)
> 
>  pub fn manage_connect(config_path: &str, rx: StdReceiver<DebouncedEvent>, task_send: Sender<Task>) {

I see.

**Pull request merged**